Source/WebCore/inspector/agents/InspectorCSSAgent.h

```diff
@@ -123,11 +124,22 @@ class InspectorCSSAgent final : public InspectorAgentBase , public Inspector::CS
     void activeStyleSheetsUpdated(Document&);
     bool forcePseudoState(const Element&, CSSSelector::PseudoClassType);
     void didChangeRendererForDOMNode(Node&);
+    void didAddEventListener(EventTarget&);
+    void willRemoveEventListener(EventTarget&);
 
     // InspectorDOMAgent hooks
     void didRemoveDOMNode(Node&, Inspector::Protocol::DOM::NodeId);
     void didModifyDOMAttr(Element&);
 
+    enum class LayoutFlag : uint8_t {
+        Rendered = 1 << 0,
+        Flex = 1 << 1,
+        Grid = 1 << 2,
+        Event = 1 << 3,
```

Comment:
this is an optimization to avoid issues on webpages that frequently modify event listeners
> Keep track of the layout flags for each `Node` to avoid re-dispatching the same flags to the frontend. This is especially important for event listeners as they can be very noisy (211169@main).